network/nodekeeper/nodekeeper.go

```diff
@@ -17,20 +17,265 @@
 package nodekeeper
 
 import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"hash"
+	"sort"
+	"sync/atomic"
+	"time"
+
+	"github.com/anacrolix/sync"
 	"github.com/insolar/insolar/core"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/sha3"
 )
 
 type NodeKeeper interface {
```

Comment:
I think u have to add an func for add active node

Reply:
+